cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -1631,6 +1631,17 @@ if(tmva-sofie)
       endif()
     endif()
   endif()
+
+  message(STATUS "Looking for BLAS for TMVA SOFIE")
+  find_package(BLAS)
+  if(NOT BLAS_FOUND)
+    if(GSL_FOUND)
+      message(STATUS "Using GSL CBLAS for TMVA SOFIE")
+    else()
+      set(tmva-sofie OFF CACHE BOOL "Disabled because BLAS was not found (${tmva-cpu_description})" FORCE)
```

Comment:
This breaks the `fail-on-missing=ON` contract. You would need a separate code path for that like in the other cases where we automatically switch off ROOT features.

Reply:
Well, it was already a problem is the code for the `tmva-cpu` option: https://github.com/root-project/root/blob/master/cmake/modules/SearchInstalledSoftware.cmake#L1665

Either this is fixed in both cases, or you keep it as it is now for consistency. But I really like the `fail-on-missing` flag so I argue for the former :)

Reply:
I think we have alread the tmva-cpu flag checking for BLAS. We need to be sure tmva-sofie is dependent on that flag, I think there is no need to another BLAS check

Reply:
Actually we can still build SOFIE without BLAS, so no need to switch OFF tmva-sofie. We need only to protect tutorials with tmva-cpu. I can make a PR for this

Reply:
Are we really okay with users that write code that is something 'like' the tutorials to "only" see an error/failure that is 'hard' to understand? I.e. the current symptom is "missing symbol _sgemm" rather than "Please rebuild ROOT once you configured it to find a BLAS library".  Can we upgrade the code to have a better error?

Reply:
"Missing symbol _sgemm" is not so hard. It is a well known issue, and asking chatGPT or Google you get 100% the right answer. And you don't need to rebuild ROOT, but just install BLAS to have the tutorial working.

Reply:
> And you don't need to rebuild ROOT, but just install BLAS to have the tutorial working.

I would still think that the error should say that rather than just `missing _sgemm`.  (Even if you figure out that you need BLAS, you probably wont know you don't need to rebuild ROOT).

> We need only to protect tutorials with tmva-cpu. I can make a PR for this

Can you please go ahead and make that PR?